spec/ics-23-vector-commitments/README.md

```diff
@@ -0,0 +1,250 @@
+---
+ics: 23
+title: Vector Commitments
+stage: draft
+category: ibc-core
+author: Christopher Goes <cwgoes@tendermint.com>
+created: 2019-04-16
+modified: 2019-04-25
+---
+
+# Synopsis
+
+A *vector commitment* is a construction that produces a succinct, binding commitment to an indexed vector of elements and short membership and/or non-membership proofs for any indicies & elements in the vector.
+This specification enumerates the functions and properties required of commitment constructions used in the IBC protocol. In particular, commitments utilized in IBC are required to be *positionally binding*: they must be able to prove existence or
+nonexistence of values at specific positions (indices).
+
+# Specification
+
+## Motivation
+
+In order to provide a guarantee of a particular state transition having occurred on one chain which can be verified on another chain, IBC requires an efficient cryptographic construction to prove inclusion or non-inclusion of particular values at particular keys in state.
+
+## Definitions
+
+The *manager* of a vector commitment is the actor with the ability and responsibility to add or remove items from the commitment. Generally this will be the state machine of a blockchain.
+
+The *prover* is the actor responsible for generating proofs of inclusion or non-inclusion of particular elements. Generally this will be a relayer (see [ICS 18](../ics-18-relayer-algorithms)).
+
+The *verifier* is the actor who checks proofs in order to verify that the manager of the commitment did or did not add a particular element. Generally this will be an IBC handler running on another chain.
+
+Commitments are instantiated with particular *key* and *value* types, which is assumed to be arbitrary serializable data.
+
+## Desired Properties
+
+This document only defines desired properties, not a concrete implementation — see "Properties" below.
+
+## Technical Specification
+
+### Datatypes
+
+An commitment construction MUST specify the following datatypes, which are otherwise opaque (need not be introspected) but must be serializable:
```

Comment:
Not sure but should this be consistently capiatlized?
```suggestion
An commitment construction MUST specify the following datatypes, which are otherwise opaque (need not be introspected) but MUST be serializable:
```

Reply:
aye, updated throughout